site/src/components/pages/docs/search/index.tsx

```diff
@@ -127,10 +130,17 @@ export const Search: React.VoidFunctionComponent<SearchProps> = ({
   };
 
   useEffect(() => {
-    if (variant === "desktop") {
+    const items = document.querySelectorAll(
+      ".searchItem",
+      // eslint-disable-next-line no-undef
+    ) as NodeListOf<HTMLButtonElement>;
```

Comment:
This isn't a big deal, and there is nothing incorrect about this approach, but I don't think it's best practice to be using query selectors like this in react.

Instead you could use store a list of refs in a `useRef`, which has the benefits of:
- creating an explicit dependency between the child components and this search component by passing the `ref` to the components through their props, rather than relying on them having the `searchItem` class name. If someone accidentally removes `searchItems` class name this would not be caught in CI, whereas if someone were to remove the `ref` passed through props it'd be a type error.
- being more type safe, as the ref would have type `HTMLDivElement | HTMLInputElement` and can only be passed to those elements. That would make the contents of `items` here reliably the elements you were looking for.

Reply:
Addressed in e164be2df807ca3043eed8ea39769ac52cd18feb.
I really dislike all the type checking that I've had to add inside the ref callback in `SearchListItem` and `SearchListHeading`, but I still prefer this approach than using `document.querySelectorAll`. If you can think of anyway to make that piece of code cleaner let me know, please.

Reply:
Thanks for addressing, since you asked how I would clean it up further I've created a small PR off this branch where you can take a look how I would've done it: https://github.com/blockprotocol/blockprotocol/pull/402

If you're happy with it, feel free to approve it and merge it into this PR.

Reply:
Addressed in https://github.com/blockprotocol/blockprotocol/pull/396/commits/89a3313f4018833368aeda52de4d6183cf96b4fe

Reply:
Thank you, It's merged